youtube_dl/extractor/americastestkitchen.py

```diff
@@ -64,3 +65,62 @@ def _real_extract(self, url):
             'series': try_get(episode, lambda x: x['show']['title']),
             'episode': episode.get('title'),
         }
+
+
+class AmericasTestKitchenPlaylistIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?americastestkitchen\.com/episodes/browse/season\_(?P<id>\d+)'
+    _TESTS = [{
+        # Single-digit season
+        'url': 'https://www.americastestkitchen.com/episodes/browse/season_1',
+        'info_dict': {
+            'id': 'season-1',
+            'title': 'Season 1',
+        },
+        'playlist_count': 13,
+    }, {
+        # Multi-digit season
+        'url': 'https://www.americastestkitchen.com/episodes/browse/season_20',
+        'info_dict': {
+            'id': 'season-20',
+            'title': 'Season 20',
+        },
+        'playlist_count': 26,
+    }]
+
+    def _real_extract(self, url):
+        mobj = re.match(self._VALID_URL, url)
+        season = mobj.group('id')
+
+        data = {
+            'requests': [
+                {
+                    'indexName': 'everest_search_atk_season_desc_production',
+                    'params': 'filters=search_document_klass%3Aepisode%20AND%20search_site_list%3Aatk%20AND%20search_show_slug%3Aatk%20&enableRules=false&hitsPerPage=30&maxValuesPerFacet=21&page=0&facets=%5B%22search_season_list%22%5D&tagFilters=&facetFilters=%5B%5B%22search_season_list%3ASeason%20' + season + '%22%5D%5D'
+
+                }
+            ]
+        }
+
+        # ATK uses a third-party provider to list the episodes for each season
+        # The data to provide it must be sent with content-type=application/x-www-form-urlencoded
+        # But the endpoint actually just wants JSON
+        season_search = self._download_json(
+            'https://y1fnzxui30-dsn.algolia.net/1/indexes/*/queries?x-algolia-agent=Algolia%20for%20JavaScript%20(3.35.1)%3B%20Browser%3B%20JS%20Helper%20(3.1.1)%3B%20react%20(16.13.1)%3B%20react-instantsearch%20(6.5.0)&x-algolia-application-id=Y1FNZXUI30&x-algolia-api-key=8d504d0099ed27c1b73708d22871d805',
```

Comment:
- use the more simpler agolia Search index endpoint.
- use `query` argument.
- fetch only the attributes needed for extraction.

Reply:
I tried a bunch of different variations and the `query` argument wasn't returning the right results. I've settled on using [facetFilters](https://www.algolia.com/doc/api-reference/api-parameters/facetFilters/) and it works well in practice